projects/vdk-core/src/vdk/api/plugin/connection_hook_spec.py

```diff
@@ -68,6 +72,70 @@ def db_connection_decorate_operation(
         """
         pass
 
+    @hookspec(firstresult=True)
+    def db_connection_execute_operation(
+        self, execution_cursor: ExecutionCursor
+    ) -> Optional[ExecuteOperationResult]:
+        """
+        The method that executes the actual SQL query using the raw_cursor.
+
+        The default implementation is basically this:
+        .. code-block::
+
+            if parameters:
+                return raw_cursor.execute(operation, parameters)
+            return raw_cursor.execute(operation)
```

Comment:
wondering, if there is a default implementation pasted within the API spec,
shouldn't it be placed directly within the spec file then, for visibility?
And yet, no implementation is a part of the API, so .. why is the API concerned with any default implementations?

My point is, if any such explanation is needed, maybe the hook(s) itself should be before or after the execute query implementation, depending on when the user wants to intercept the workflow.
A before-query-execution hook is validate & decorate hooks. What was missing, is after-query-execution hook, is that the problem solved by the PR?